youtube_dl/extractor/soundcloud.py

```diff
@@ -469,3 +472,70 @@ def _real_extract(self, url):
             'description': data.get('description'),
             'entries': entries,
         }
+
+
+class SoundcloudSearchIE(SearchInfoExtractor, SoundcloudIE):
+    IE_NAME = 'soundcloud:search'
+    IE_DESC = 'Soundcloud search'
+    _MAX_RESULTS = 200
```

Comment:
>The number of items in the collection returned is limited to 50 by default with a maximum value of 200

Collection is a single page thus 200 is the limit for a single page. So, `_RESULTS_PER_PAGE` should be set to 200 and `_MAX_RESULTS` to `float('inf')`.